listeners.go

```diff
@@ -146,25 +158,55 @@ func updateListeners(port string) error {
 		vanished[host] = false
 	}
 	for _, host := range hosts {
-		if _, ok := listeners[host]; ok {
+		if servers, ok := listeners[host]; ok {
 			// confirm found
 			delete(vanished, host)
-			continue
+			cont := false
+			for _, server := range servers {
+				if server.port == port {
+					cont = true
+					break
+				}
+			}
+			if cont {
+				continue
+			}
 		}
 		addr := net.JoinHostPort(host, port)
 		ln, err := net.Listen("tcp", addr)
 		if err != nil {
 			log.Println(err) // err includes enough context
+			log.Println("(error on listen)")
```

Comment:
So apparently err didn’t include enough context after all :)

Let’s make this `log.Printf("listen: %v", err)`, then?